snakemake/deployment/__init__.py

```
@@ -1,0 +1,89 @@
+        for kind, plugin in self.plugins.items():
+            if kind in global_variables:
+                raise WorkflowError(
+                    "The software deployment plugin {plugin.name} provides an "
+                    "invalid kind {kind}, which is already reserved as a global "
+                    "variable in snakemake."
+                )
+
+        for kind, plugin in self.plugins.items():
+
+            def factory(*args, within: EnvSpecBase, **kwargs):
+                env_spec = plugin.env_spec_cls(*args, **kwargs)
```

Ruff (0.8.2)

Function definition does not bind loop variable `plugin`

(B023)